modules/scala-support/src/main/scala/dev/guardrail/generators/scala/circe/CirceProtocolGenerator.scala

```diff
@@ -28,12 +29,19 @@ class CirceProtocolGeneratorLoader extends ProtocolGeneratorLoader {
 
 object CirceProtocolGenerator {
   def apply(circeVersion: CirceModelGenerator): ProtocolTerms[ScalaLanguage, Target] =
-    new CirceProtocolGenerator(circeVersion, (tpe, _) => Target.pure(tpe))
-  def withValidations(circeVersion: CirceModelGenerator, applyValidations: (Type, Tracker[Schema[_]]) => Target[Type]): ProtocolTerms[ScalaLanguage, Target] =
+    new CirceProtocolGenerator(circeVersion, WithValidations.ignore)
+  def withValidations(circeVersion: CirceModelGenerator, applyValidations: WithValidations): ProtocolTerms[ScalaLanguage, Target] =
     new CirceProtocolGenerator(circeVersion, applyValidations)
+
+  trait WithValidations {
```

Comment:
I think a SAM interface is better from readability perspective, at least the arguments have names and one can use name = value assignment in an arbitrary order if needed making it easier to refactor

Reply:
This is a great idea.